crates/bevy_pbr/src/ssao/mod.rs

```diff
@@ -0,0 +1,958 @@
+                    view_formats: &[],
+                },
+            );
+
+            let ssao_texture = texture_cache.get(
+                &render_device,
+                TextureDescriptor {
+                    label: Some("ssao_texture"),
+                    size,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::R16Float,
+                    usage: TextureUsages::STORAGE_BINDING | TextureUsages::TEXTURE_BINDING,
+                    view_formats: &[],
+                },
+            );
+
+            let depth_differences_texture = texture_cache.get(
+                &render_device,
+                TextureDescriptor {
+                    label: Some("ssao_depth_differences_texture"),
+                    size,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::R32Uint,
+                    usage: TextureUsages::STORAGE_BINDING | TextureUsages::TEXTURE_BINDING,
+                    view_formats: &[],
+                },
+            );
+
+            commands
+                .entity(entity)
+                .insert(ScreenSpaceAmbientOcclusionTextures {
+                    preprocessed_depth_texture,
+                    ssao_noisy_texture,
+                    screen_space_ambient_occlusion_texture: ssao_texture,
+                    depth_differences_texture,
+                });
+        }
+    }
+}
+
+#[derive(Component)]
+struct SSAOPipelineId(CachedComputePipelineId);
+
+fn prepare_ssao_pipelines(
+    mut commands: Commands,
+    pipeline_cache: Res<PipelineCache>,
+    mut pipelines: ResMut<SpecializedComputePipelines<SSAOPipelines>>,
+    pipeline: Res<SSAOPipelines>,
+    views: Query<(
+        Entity,
+        &ScreenSpaceAmbientOcclusionSettings,
+        Option<&TemporalJitter>,
+    )>,
+) {
+    for (entity, ssao_settings, temporal_jitter) in &views {
+        let pipeline_id = pipelines.specialize(
+            &pipeline_cache,
+            &pipeline,
+            SSAOPipelineKey {
+                ssao_quality: ssao_settings.clone(),
+                temporal_noise: temporal_jitter.is_some(),
+            },
+        );
+
+        commands.entity(entity).insert(SSAOPipelineId(pipeline_id));
+    }
+}
+
+#[derive(Component)]
+struct SSAOBindGroups {
+    common_bind_group: BindGroup,
+    preprocess_depth_bind_group: BindGroup,
+    gtao_bind_group: BindGroup,
+    spatial_denoise_bind_group: BindGroup,
+}
+
+fn queue_ssao_bind_groups(
+    mut commands: Commands,
+    render_device: Res<RenderDevice>,
+    pipelines: Res<SSAOPipelines>,
+    view_uniforms: Res<ViewUniforms>,
+    global_uniforms: Res<GlobalsBuffer>,
+    views: Query<(
+        Entity,
+        &ScreenSpaceAmbientOcclusionTextures,
+        &ViewPrepassTextures,
+    )>,
+) {
```

Comment:
This system panics with an obscure error message if the user forgets to add `TemporalAntiAliasPlugin`. Would it be possible to check for this at the start of this system and give a nicer panic message than the following?

```
wgpu error: Validation Error

Caused by:
    In Device::create_bind_group
      note: label = `ssao_preprocess_depth_bind_group`
    Texture binding 0 expects multisampled = false, but given a view with samples = 4
```

Reply:
Ah so it's really panicking because MSAA must be off, which the TAA plugin does for you. I'll add a better error message though, yeah.